tests/Properties/ByteString.hs

```diff
@@ -284,6 +284,11 @@ tests =
     \x -> B.length x === fromIntegral (length (B.unpack x))
   , testProperty "count" $
     \(toElem -> c) x -> B.count c x === fromIntegral (length (elemIndices c (B.unpack x)))
+  -- for the long strings, the multiplier is non-round (and not power of 2)
+  -- to ensure non-trivial prefix or suffix of the string is handled outside any possible SIMD-based loop,
+  -- which typically handles chunks of 16 or 32 or 64 etc bytes.
+  , testProperty "count (long strings)" $ mapSize (* 1025) $
+    \(toElem -> c) x -> B.length x >= 1024 ==> B.count c x === fromIntegral (length (elemIndices c (B.unpack x)))
```

Comment:
I'm afraid this way a list in the right-hand size can quickly get out of hand. Imagine running tests with `--quickcheck-tests 100000`, and then these 100000 are multiplied by 1025. That's why I [was suggesting](https://github.com/haskell/bytestring/pull/202#issuecomment-786851139) a property with `replicate`.